### ISSUE TITLE
Sudoku error

### ISSUE THREAD

**Issue by RenatoGeh** [azreasoners/NeurASP#4]
Hi,

I'm trying to run the Sudoku example in `examples/sudoku`, but I ran into this error when running `python train.py`:

```
Error: none considered case for output with shape torch.Size([1, 81, 10]) v.s. label with shape torch.Size([1, 9, 9])
```

The same error comes up when running `python train.py` in `examples/solvingSudoku_70k`.

```
Error: none considered case for output with shape torch.Size([230, 81, 9]) v.s. label with shape torch.Size([230, 9, 9])
```

What am I missing?

Thanks

**Comment by zhunyoung** [comment]
Hi Renato,

Thank you for your interest in our work and for sharing the issues you encountered when running the codes. 

I looked into the issue and, in short, this is a bug in the codes and I just resolved it in commit e855ce2. You can update your codes by running `git pull` under the project folder `NeurASP` to resolve this issue.

This issue is introduced by our previous generalization in file `neurasp.py` where we removed some domain-specific shape modifications in method `NeurASP.testNN`. More specifically, the `NeurASP.testNN` method is expecting a label of shape (81) instead of (9, 9) to match with the prediction of shape (81, 10) in `examples/sudoku` or (81, 9) in `examples/solvingSudoku_70k`. What I did in commit e855ce2 is to simply add a reshaping step for the label.

Thank you again for pointing out this issue and please let us know if you have any questions.

Cheers,
Zhun

**Comment by RenatoGeh** [comment]
Working now, many thanks!
